x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/support/UserBoolQueryBuilder.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.support;
+
+import org.apache.lucene.search.Query;
+import org.elasticsearch.index.query.BoolQueryBuilder;
+import org.elasticsearch.index.query.ExistsQueryBuilder;
+import org.elasticsearch.index.query.MatchAllQueryBuilder;
+import org.elasticsearch.index.query.PrefixQueryBuilder;
+import org.elasticsearch.index.query.QueryBuilder;
+import org.elasticsearch.index.query.QueryBuilders;
+import org.elasticsearch.index.query.QueryRewriteContext;
+import org.elasticsearch.index.query.SearchExecutionContext;
+import org.elasticsearch.index.query.TermQueryBuilder;
+import org.elasticsearch.index.query.TermsQueryBuilder;
+import org.elasticsearch.index.query.WildcardQueryBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.xpack.security.support.SecurityIndexFieldNameTranslator.exact;
+
+public class UserBoolQueryBuilder extends BoolQueryBuilder {
+    public static final SecurityIndexFieldNameTranslator USER_FIELD_NAME_TRANSLATOR = new SecurityIndexFieldNameTranslator(
+        List.of(exact("username"), exact("roles"), exact("metadata"), exact("full_name"), exact("email"), exact("enabled"))
+    );
+
+    private UserBoolQueryBuilder() {}
+
+    public static UserBoolQueryBuilder build(QueryBuilder queryBuilder) {
+        UserBoolQueryBuilder userQueryBuilder = new UserBoolQueryBuilder();
+        if (queryBuilder != null) {
+            QueryBuilder translaterdQueryBuilder = translateToUserQueryBuilder(queryBuilder);
+            userQueryBuilder.must(translaterdQueryBuilder);
+        }
+        userQueryBuilder.filter(QueryBuilders.termQuery("type", "user"));
```

Comment:
For apikey this is called `doc_type`, just wanted to confirm that `type` for user is correct and expected here?

Reply:
Thank you for doing that! Looks like this filter is acceptable based on the conversation and your comment here.